cli/azd/test/recording/poll_discard.go

```diff
@@ -0,0 +1,363 @@
+package recording
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"gopkg.in/dnaeon/go-vcr.v3/cassette"
+)
+
+// httpPollDiscarder discards awaiting-done polling interactions from the cassette.
+// As a result of this, the cassette will only contain the final result of the polling operation.
+// When playing back interactions from the cassette, this results in fast-forwarding behavior.
+//
+// The type of polling protocols httpPollDiscarder detects includes:
+// - Azure specific async polling protocols (implementation closely matches ones supported by azure-sdk-for-go)
+// - Idiomatic HTTP async polling protocols (Location header, 201,202 status codes)
+type httpPollDiscarder struct {
+	pollInProgress poller
+}
+
+func (d *httpPollDiscarder) BeforeSave(i *cassette.Interaction) error {
+	if d.pollInProgress != nil && d.pollInProgress.Done(i) {
```

Comment:
Just wondering - it seems like if we have two concurrent polling requests to different resources this logic will end up not dropping some intermediate values (because of how the requests are interleaved).  Should we have some dictionary here keyed by request URL of pollers we are tracking?

Reply:
Seems like a simple enough change to pull in into scope, I'll make it happen!

I was mostly leaving a naive implementation and expanding it as necessary as we need it. There will certainly be hiccups in rough areas where we'll need to add functionality.

Reply:
I've implemented concurrent handling with some basic tests.

As you might have seen, the current implementation drops requests not relevant to the polling operation while any ongoing polling operation is in progress. This achieved the behavior of handling azd's async deployment mechanism by default, but the hack here ends up being undesirable long term, which is made clear with the additional concurrency support here.

I'm going to do another pass to change it to only drop pertinent polling operation requests, and disable the UI refresh via an env var, but let me know if you have any other thoughts.

Reply:
I've made the changes, and do like how it looks in the end. I do think this ends up fixing the idiosyncrasies and making a functionally correct system.